docs/advanced-topics/negative-mining.md

```diff
@@ -0,0 +1,91 @@
+(negative-mining)=
+# {octicon}`telescope` Negative Mining
+
+Negative Mining is an advanced technique, which optimizes the way data is sampled from you training dataset.
+Usually, it aims at making the metric learning tasks for the model harder during the training. 
+In this way it can lead to better fine-tuning results.
+
+## Context: Deep Metric Learning
+
+To understand negative mining, we first take a look at how the training data 
+construction is done for metric learning tasks.
+
+Metric Learning algorithms attempt to teach neural network models to tell
+which objects are semantically/visually similar and which ones are not.
+
+For uni-modal fine-tuning tasks such as text-to-text, image-to-image, or mesh-to-mesh,
+Finetuner constructs training data in the following way:
+
+![batch-sample](../imgs/batch-sampling.png)
+
+Assume we have a list of Documents belonging to four classes: `1`, `2`, `3`, and `4`,
+Finetuner will evenly sample *X* items per class to make a batch *B*.
+
+In the next step, arranges the items.
+Many loss functions used by Finetuner are calculated based on the embeddings of Triples. 
+For those, Finetuner creates all possible Triplets *(anchor, pos, neg)* from this batch.
+These Triplets become the training data.
+The objective is to pull Documents that belong to the same class together,
+while pushing the Documents which belong to a different class away from each other.
+
+![training](../imgs/metric-train.png)
+
+
+## The Triplet Margin Miner
+
+After Triplets construction,
+all possible *(anchor, pos, neg)* Triplets becomes the training data.
+However, what if the pre-trained model already performs well on some of the Triplets?
+
+To put it in another way: *what if the distance between `anchor` and `pos` is much smaller than
+the distance between `anchor` and `neg`?*
+
+These Triplets, does not contribute to improving the model.
```

Comment:
```suggestion
These Triplets do not contribute to improving the model, since they are already in the desired relation to each other in the embedding space.
```